amz_stream_infra/stack_definitions_firehose.py

```diff
@@ -0,0 +1,181 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import aws_cdk
+from constructs import Construct
+import json
+import os
+from aws_cdk import (
+    Environment,
+    Fn as fn,
+    Duration,
+    Stack,
+    Tags,
+    CfnOutput,
+    aws_iam as iam,
+    aws_sqs as sqs,
+    aws_sns as sns,
+    aws_lambda as _lambda,
+    aws_lambda_event_sources as lambda_events,
+    aws_s3 as s3,
+    aws_kinesisfirehose_alpha as firehose,
+    aws_kinesisfirehose_destinations_alpha as destinations,
+    aws_lambda_event_sources as lambda_event_source
+)
+
+class DataSetScopedConstruct(Construct):
+    """
+    Base construct which has scoped to dataset
+    """
+    def __init__(self, scope: Construct, construct_id: str, ambassadors_config, dataset_config):
+        super().__init__(scope, construct_id)
+        self.ambassadors_config = ambassadors_config
+        self.dataset_config = dataset_config
+
+#Creating Subscription role and its policies  
+class AmzStreamFirehoseSubscriptionRoleInfra(DataSetScopedConstruct):
+    def __init__(self, scope: Construct, construct_id: str, ambassadors_config, dataset_config, firehose: firehose.DeliveryStream,advertising_region) -> None:
+        super().__init__(scope, construct_id, ambassadors_config, dataset_config)
+
+        self.firehose_subscription_role = iam.Role(
+            self, "FirehoseSubscriptionRole",
+            role_name="-".join([dataset_config['dataSetId'][:8],advertising_region,"subscription"]),
```

Comment:
can we increase the char limit of the dataset name? IAM role name has 64 chars and some of our dataset names are much longer than 8